runtime/compilers/rillv1beta/connector.go

```diff
@@ -61,7 +62,7 @@ func ExtractConnectors(ctx context.Context, projectPath string) ([]*Connector, e
 	}
 
 	// keeping a map to dedup connectors
-	connectorMap := make(map[key]bool)
+	connectorMap := make(map[key][]string)
```

Comment:
This will depend on the question raised below :
If different connectors use different field the logic should be driven by `Spec` and the logic to understand `Spec` should be here.